lua-plugins/cli_filter.lua.in

```diff
@@ -256,14 +318,28 @@ function slurm_cli_pre_submit(options, offset)
         if pinfo == nil then return slurm_error("unable to retrieve partition information") end
 
         local tres = pinfo.TRES
-        if not tres or not tres.cpu or not tres['gres/gpu'] then return slurm_error('unable to determine cpu to gpu ratio') end
+        if not tres or not tres.cpu or not tres['gres/gpu'] or not tonumber(tres['gres/gpu']) then
+            return slurm_error('unable to determine cpu to gpu ratio')
+        end
+
         local cpus_per_gpu = tonumber(tres.cpu)/tonumber(tres['gres/gpu'])
         if tonumber(options['threads-per-core']) == 1 then cpus_per_gpu = cpus_per_gpu/2 end
 
         if has_explicit_cpu_request and not is_srun_in_allocation then
             return slurm_errorf('cannot explicitly request CPU resources for GPU allocation; each allocated GPU allocates %d cores', cpus_per_gpu)
         end
 
+        -- In principle, can use sinfo -o %G to get node configured gres; this information is not in pinfo.TRES.
+        -- For now, in order to address NVMe tmp allocation issues on the gpu partitions, the capacities are hard-coded:
+        --     Allocatable NVMe on tmp is 3500 GiB
+        --     Default allocation, set in job_submit.lua, is 128 GiB.
+
+        local gpus_per_node = 8
```

Comment:
This could be an issue as gpus_per_node=8 for the AMD gpus but not the GH200 nodes, which have 4.